packages/replay/src/main.ts

```diff
@@ -334,28 +335,38 @@ async function doUploadRecording(
   const metadata = await validateMetadata(client, recording.metadata, verbose);
 
   let recordingId: string;
-  if (size > MIN_MULTIPART_UPLOAD_SIZE && process.env.REPLAY_MULTIPART_UPLOAD) {
-    recordingId = await multipartUploadRecording(
-      server,
-      client,
-      dir,
-      recording,
-      metadata,
-      size,
-      strict,
-      verbose
-    );
-  } else {
-    recordingId = await directUploadRecording(
-      server,
-      client,
-      dir,
-      recording,
-      metadata,
-      size,
+  try {
+    if (size > MIN_MULTIPART_UPLOAD_SIZE && process.env.REPLAY_MULTIPART_UPLOAD) {
+      recordingId = await multipartUploadRecording(
+        server,
+        client,
+        dir,
+        recording,
+        metadata,
+        size,
+        strict,
+        verbose
+      );
+    } else {
+      recordingId = await directUploadRecording(
+        server,
+        client,
+        dir,
+        recording,
+        metadata,
+        size,
+        strict,
+        verbose
+      );
+    }
+  } catch (err) {
+    handleUploadingError(
+      err instanceof ProtocolError ? err.protocolMessage : String(err),
       strict,
-      verbose
+      verbose,
+      err
     );
+    return null;
```

Comment:
The name `handleUploadingError` is misleading. Check out more context here: https://github.com/replayio/replay-cli/pull/302. In essence, it introduced a strict mode causing the process to fail on any error, even if we proceeded with some errors (like failing to upload a sourcemap). Irrespective of the strict flag, there's no point continuing if we fail to upload the recording binary.

Reply:
Thanks for the review and the pointers! I see what you are saying about `handleUploadingError` and the `strict`'s flag purpose etc. This one is still treated as a fatal error though since we immediately `return null`. It's just like you are saying - there is no point continuing if this fails (or well, we could retry or something on some specific error codes - but other than the one that the ticker is about because when the limit of per API key recordings has been reached then there is truly no point in continuing here).

I based this logic on the code that handles `initConnection` failure:
https://github.com/replayio/replay-cli/blob/1e6b7ec1a9629764b2be24760ab4e3e7ef6abb87/packages/replay/src/main.ts#L328-L331

I think they are mostly the same in terms of their severity etc. If this is already handled like this here - is there a point in hoisting this added error handler to one of the callers? One advantage of reusing `handleUploadingError` is that we automatically get a nice prefix string (`Upload failed: `) for the logged error (it's logged based on the `verbose` flag but from what I can see it's basically - almost - always set to `true` anyway).

Let me know what do you think, I don't mind hoisting it if you think it would be better.

Reply:
Ah, I missed the return null part.